index.js

```diff
@@ -0,0 +1,82 @@
+const Base = require('bfx-facs-base')
+const crypto = require('crypto')
+const fs = require('fs/promises')
+const path = require('path')
+const TTLCache = require('@isaacs/ttlcache')
+
+class TTLCacheFs extends Base {
+  constructor (caller, opts = {}, ctx) {
+    super(caller, opts, ctx)
+
+    this.name = 'ttlcache_fs'
+    this._hasConf = false
+
+    this._cacheDirectory = opts.directory ?? './cache/data'
+
+    if (opts?.deleteCacheDirOnStart ?? true) {
+      // Clear cache directory on start
+      fs.rm(this._cacheDirectory, {
+        force: true,
+        recursive: true
+      }).catch(err => console.trace(`Could not clear cache directory ${this._cacheDirectory}`, err))
+    }
+
+    const cacheOpts = {
+      max: 1000,
+      ttl: 3600 * 1000,
+      dispose: (value, key) => {
+        const { filePath } = this._getKeyFilenameInCache(key)
+        fs.rm(filePath, { force: true }).catch(err => console.trace(`Could not remove cached file for key ${key}`, err))
+      },
+      ...opts?.cache
+    }
+
+    this._cache = new TTLCache(cacheOpts)
+
+    this.init()
+  }
+
+  _getKeyFilenameInCache (key) {
```

Comment:
Should we add a null check of key here?

Reply:
I thought about this originally but then figured this does not throw on `undefined` or `null` and in case we would want to handle these cases we would throw.

I think it's better not to throw and leave it up to the users of the facility to properly construct keys being passed in.